### PR TITLE
Add support for `tuple` and `set` in intrinsic `type(object)`

### PR DESCRIPTION
**Pull request by kmr-srbh**
```python
my_tuple: tuple[str, i32] = ("A", 1)
my_set: set[str] = {"A", "B", "C"}

print(type(my_tuple))
print(type(my_set))
```
```console
(lp) saurabh-kumar@Awadh:~/Projects/System/lpython$ ./src/bin/lpython ./examples/example.py
<class 'tuple'>
<class 'set'>
```

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
It looks like there is a failure at the CI.

**Comment by kmr-srbh** [comment]
> It looks like there is a failure at the CI.

Yes @Shaikh-Ubaid. It seems we are having empty printf("") statements in the generated code for C back-end. The issue occurs when printing list and tuple.

**Comment by kmr-srbh** [comment]
Removed the C back-end from tests as `set` is not implemented for it.

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -831,7 +831,7 @@ RUN(NAME callback_04       IMPORT_PATH .. LABELS cpython)
 # Intrinsic Functions
 RUN(NAME intrinsics_01     LABELS cpython llvm llvm_jit NOFAST) # any
 RUN(NAME intrinsics_02     LABELS cpython llvm llvm_jit c) # floordiv
-RUN(NAME test_builtin_type LABELS cpython llvm llvm_jit c) # type
+RUN(NAME test_builtin_type LABELS cpython llvm llvm_jit) # type
```
Comment (Shaikh-Ubaid) [inline integration_tests/CMakeLists.txt:834]: @kmr-srbh we do not remove any backends for existing tests (unless extremely necessary). If there is a feature not supported by any of the previously enabled backends, then we add a new/separate integration test.

Comment (kmr-srbh) [reply integration_tests/CMakeLists.txt:834]: @Shaikh-Ubaid I removed the test for it due to https://github.com/lcompilers/lpython/pull/2687#issuecomment-2096218282

Comment (Shaikh-Ubaid) [reply integration_tests/CMakeLists.txt:834]: Just add the set tests in a new integration test and for it enable the other supporting backends (like llvm, llvm_jit, etc.). That way, we do not stop testing any existing tests for the C (or other) backends.

Comment (kmr-srbh) [reply integration_tests/CMakeLists.txt:834]: > Just add the set tests in a new integration test and for it enable the other supporting backends (like llvm, llvm_jit, etc.). That way, we do not stop testing any existing tests for the C (or other) backends.

Got it @Shaikh-Ubaid! I will remember this for future commits. :+1:

File: integration_tests/test_builtin_type_set.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from lpython import i32
+
+def test_builtin_type_set():
+    st: set[i32] = {1, 2, 3, 4}
+
+    res = str(type(st))
```
Comment (Shaikh-Ubaid) [inline integration_tests/test_builtin_type_set.py:6]: Looks like there is bug here:

```console
semantic error: Variable 'res' not declared
 --> /Users/runner/work/lpython/lpython/integration_tests/test_builtin_type_set.py:6:5
  |
6 |     res = str(type(st))
  |     ^^^ 


Note: Please report unclear or confusing messages as bugs at
https://github.com/lcompilers/lpython/issues.
make[2]: *** [test_builtin_type_set.o] Error 2
make[1]: *** [CMakeFiles/test_builtin_type_set.dir/all] Error 2
make: *** [all] Error 
```

File: integration_tests/test_builtin_type.py [hunk 3]
```diff
@@ -6,6 +6,8 @@ def test_builtin_type():
     s: str = "Hello, LPython!"
     l: list[i32] = [1, 2, 3, 4, 5]
     d: dict[str, i32] = {"a": 1, "b": 2, "c": 3}
+    t: tuple[str, i32] = ("a", 1)
+    st: set[i32] = {1, 2, 3, 4}
```
Comment (Shaikh-Ubaid) [inline integration_tests/test_builtin_type.py:10]: You will need to remove this `set` test from here.

Comment (kmr-srbh) [reply integration_tests/test_builtin_type.py:10]: I am really sorry for the above mistakes @Shaikh-Ubaid. I was in a hurry and pushed unfinished changes. I am fixing this.

**Review by Shaikh-Ubaid — APPROVED** [review]

**Pull request merged**